servicetalk-concurrent-api/src/testFixtures/java/io/servicetalk/concurrent/api/TestCompletable.java

```diff
@@ -74,6 +77,14 @@ public boolean isSubscribed() {
         return !(subscriber instanceof WaitingSubscriber);
     }
 
+    /**
+     * Awaits until this {@link TestCompletable} is subscribed, even if interrupted. If interrupted the
+     * {@link Thread#isInterrupted()} will be set upon return.
+     */
+    public void awaitSubscribed() {
```

Comment:
Can you please elaborate the use-case? `subscribe` is a synchronous operation. Usually this await is not required because the source is subscribed as soon as you do `subscribe`. Is it for the case when you initiate subscribe from another thread?

Reply:
The `subscribe()` is being offloaded because of the `subscribeOn()` operator. The thread which initiate the `subscribe()` regains control before the `subscribe` completes on the offloaded thread. `awaitSubscribed()` allows it to wait until the subscribe is completed.